babi.py

```diff
@@ -914,7 +914,82 @@ def _get_char(stdscr: 'curses._CursesWindow') -> Key:
     return Key(wch, key, keyname)
 
 
-EditResult = enum.Enum('EditResult', 'EXIT NEXT PREV')
+EditResult = enum.Enum('EditResult', 'EXIT NEXT PREV EDIT')
+
+
+class Command:
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        keybinds: List[str],
+        func: Callable[[Screen, Key], EditResult],
+        aliases: List[str] = [],
+    ) -> None:
+        self.name = name
+        self.description = description
+        self.keybinds = keybinds
+        self.aliases = aliases
+        self.func = func
+
+    def translate_keybinds(self) -> List[bytes]:
+        """
+        Translates the keybinds in the form "Modifier-key" to curses-compatible
+        format.
+        """
+        new_binds = []  # type: List[bytes]
+        for bind in self.keybinds:
+            new_binds.append(bind.replace('C-', '^').upper().encode('UTF-8'))
+        return new_binds
```

Comment:
I would leave this out, no need to fight the programmer who could just write `^` instead of `C-`

Reply:
Done in a commit localle